### PR TITLE
OPENEUROPA-2502: Add param groups to validation path.

### PR DESCRIPTION
**Pull request by dxvargas**
### Description

Add param groups to validation path to get user groups.

### Change log

- Added: Parameter groups=* in `\Drupal\oe_authentication\Event\EuLoginEventSubscriber::alterValidationPath`

### CODE REVIEW

File: src/CasProcessor.php [hunk 1]
```diff
@@ -87,7 +87,12 @@ private static function parseAttributes(\DOMElement $node): array {
       else {
         $value = $child->nodeValue;
       }
-      $attributes[$name] = $value;
+      if ($name === 'group') {
```
Comment (upchuk) [inline src/CasProcessor.php]: Can't we figure out if the attribute is an array without hardcoding the name of it?  Also, isn't the `group` attribute one array level down?

Also, no tests for this?

File: src/CasProcessor.php [hunk 2]
```diff
@@ -72,31 +72,60 @@ public static function processValidationResponseAttributes(string $source): arra
...
+   */
+  protected static function isAssociative(\DOMElement $node): bool {
+    $names = [];
+    foreach ($node->childNodes as $key => $child) {
+      $names[] = $child->localName;
+    }
+
+    // We consider it as associative if we have only one name value or if  the
+    // name values are different.
+    return count($names) < 2 || count(array_unique($names)) > 1;
```
Comment (imanoleguskiza) [inline src/CasProcessor.php]: Not sure if this is possible but if we have some children with the same name and some others with different ones then they will be processed as associative and the children with the same name will be overwriten.

File: src/CasProcessor.php [hunk 3]
```diff
@@ -72,31 +72,60 @@ public static function processValidationResponseAttributes(string $source): arra
...
+   * @return bool
+   *   Whether the node children should be mapped to an associated array.
+   */
+  protected static function isAssociative(\DOMElement $node): bool {
+    $names = [];
+    foreach ($node->childNodes as $key => $child) {
+      $names[] = $child->localName;
+    }
+
+    // We consider it as associative if we have only one name value or if  the
```
Comment (imanoleguskiza) [inline src/CasProcessor.php]: Double space before the "the"

File: src/CasProcessor.php [hunk 4]
```diff
@@ -72,31 +72,60 @@ public static function processValidationResponseAttributes(string $source): arra
...
+   *
+   * @param \DOMElement $node
+   *   The node element.
+   *
+   * @return bool
+   *   Whether the node children should be mapped to an associated array.
+   */
+  protected static function isAssociative(\DOMElement $node): bool {
+    $names = [];
+    foreach ($node->childNodes as $key => $child) {
```
Comment (imanoleguskiza) [inline src/CasProcessor.php]: No need to define the $key variable

File: src/CasProcessor.php (hunk 2 above)
Comment (dxvargas) [reply src/CasProcessor.php]: Fixed.

**Pull request merged**